### ISSUE TITLE
Feature request: please add TOC to documentation site

### ISSUE THREAD

**Issue by lonix1** [Microsoft/TypeScript#29606]
## Search Terms
handbook, typescriptlang, website

## Suggestion

The documentation on the website ([example](https://www.typescriptlang.org/docs/handbook/basic-types.html)) if not user friendly at all.

There is a TOC on the left for all the pages. But there is no TOC per-page.

Scroll down that example I linked to - see how hard it is to find any info? There needs to be a TOC at the top of the page, and even better, a scrolling TOC on the right that shows where you are right now.

The documentation is of a high quality, but accessing it is a chore.

**Comment by dragomirtitian** [comment]
This would also help a lot with linking. Right now if I want to point someone to the documentation for let's say 'the distributive behavior of conditional types' I can only link to the advanced types page, which is **long**, and while the info is there a direct link to the section would be better.

**Comment by fatcerberus** [comment]
Yeah, I always expect the headers to have magic permalink anchors (like GitHub markdown) but then I hover over them and no anchor link appears.  Then I get sad. 😢

**Comment by orta** [comment]
Still figuring out if we'll do the TOC on a per-page basis in the new site, but the permalink are in 👍
